cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -233,13 +256,57 @@ private DatasetDefinitionRegistry createRegistry(@Nullable ClassLoader classLoad
     return registry;
   }
 
+  private List<String> getAvailableModuleClasses(Id.Namespace namespace) {
+    LinkedList<String> allClasses = getAllModuleClasses(systemNamespace);
+    if (systemNamespace.equals(namespace)) {
+      return allClasses;
+    }
+    LinkedList<String> namespaceClasses = getAllModuleClasses(namespace);
+    for (String namespaceClass : namespaceClasses) {
+      int idx = allClasses.indexOf(namespaceClass);
+      if (idx > -1) {
```

Comment:
This is applies when you have the same type class in the system namespace as well as the current module's namespace. In such a scenario, when you reach here, ``allClasses`` has type classes from modules in the system namespace. While iterating over the type classes from modules in the current namespace, if you encounter a class that is already present in ``allClasses``, you remove it, then add the corresponding version of the class from the module in the current namespace. This is to give type classes in the current namespace precedence over type classes in the system namespace.